### ISSUE TITLE
Add onException(e: Exception) to streaming handler for detecting network disconnection

### ISSUE THREAD

**Issue by MulticolorWorld** [sys1yagi/mastodon4j#52]

**Comment by MulticolorWorld** [comment]
## idea 1
Continue to recconect streaming until called shutdown()

    fun connectHoge(handler: Handler): Shutdownable {
        val dispatcher = Dispatcher()
        dispatcher.invokeLater(Runnable {
            while (true) {
                val response = client.get("hoge")
                if (response.isSuccessful) {
                    val reader = response.body().byteStream().bufferedReader()
                    try {
                        while (true) {
                            val line = reader.readLine()
                            .....
                        }
                    } catch (e: IOException) {
                        //waiting reconnect time
                        continue
                    }
                }
            }
        })
        return Shutdownable(dispatcher)
    }

**Comment by sys1yagi** [comment]
I agree the idea. But I think that Streaming shouldn't have responsibility for automatic reconnection.
It would be better to provide a disconnect event and a reconnect function.

```kotlin
interface Retryable {
    fun retry()
}

interface Handler {
    // ...
    fun onDisconnected(retryable: Retryable)
}
```

It can delegate the reconnection strategy to the user.

**Comment by MulticolorWorld** [comment]
OK. I try to write code according to this sample. Thank you!
